2017/Team-Training/OMERO-web/index.html

```diff
@@ -0,0 +1,566 @@
+                    
+                    Listen for and trigger events
+                    <pre><code class="javascript">// double-click handler on image. NB: event bound to parent container
+$("#content_details").on("dblclick", "li.row", function(event) {
+...});
+
+// Listen for custom events. right_plugin.general.js.html
+$("body").on("selection_change.ome", function(event) {
+    // clear contents of right panel, then update
+    $("#metadata_general").empty();
+    update_metadata_general_tab();
+});
+
+// Trigger events. ome.webclient.actions.js
+OME.handle_tree_selection = function(data, event) {
+    ...
+    $("body").trigger("selection_change.ome", data);
+}
+                    </code></pre>
+
+                    <a target="_blank" href="https://github.com/openmicroscopy/openmicroscopy/blob/develop/components/tools/OmeroWeb/omeroweb/webclient/templates/webclient/data/includes/center_plugin.thumbs.js.html#L89">
+                        center_plugin.thumbs.js.html
+                    </a>,
+                    <a target="_blank" href="https://github.com/openmicroscopy/openmicroscopy/blob/develop/components/tools/OmeroWeb/omeroweb/webclient/templates/webclient/data/includes/right_plugin.general.js.html#L153">
+                        right_plugin.general.js.html
+                    </a>,
+                    <a target="_blank" href="https://github.com/openmicroscopy/openmicroscopy/blob/develop/components/tools/OmeroWeb/omeroweb/webclient/static/webclient/javascript/ome.webclient.actions.js#L75">
+                        ome.webclient.actions.js
+                    </a>
+                </section>
+
+
+                <section>
+                    <h2>jsTree</h2>
+
+                    Managing hierarchies
+                    <pre><code class="javascript">$("#dataTree")
+    .on('changed.jstree', function (e, data) {
+        OME.tree_selection_changed(data, e);
+    })
+    .on('move_node.jstree', function(e, data) {
+        // update links to server
+    })
+    .jstree({
+        'core': {
+            'data': function(node, callback) {
+                if (node.type === 'project') {
+                    url = WEBCLIENT.URLS.api_datasets;
+                }
+
+                $.ajax({url: url, data: payload,
+                    success: function (data, textStatus, jqXHR) {
+                        callback.call(this, data);
+                    }
+                });
+            }
+        }
+    });
+                    </code></pre>
+                    <a target="_blank" href="https://github.com/openmicroscopy/openmicroscopy/blob/develop/components/tools/OmeroWeb/omeroweb/webclient/static/webclient/javascript/ome.tree.js">
+                        /webclient/static/webclient/javascript/ome.tree.js
+                    </a>
+                </section>
+
+
+                <section>
+                    <h2>Underscore templates</h2>
+
+                    Used to generate HTML from 'JSON'
```

Comment:
Will you mention @rgozim presentation at that stage?

Reply:
Not at this point, but certainly when I'm on the "Browser - Server communication" slide I'll mention JSON generation.